rackcity/views/rack_views.py

```diff
@@ -27,32 +35,7 @@ def rack_get(request):
         row_letter__range=range_serializer.get_row_range(),
     )
 
-    if racks.count() == 0:
-        return JsonResponse(
-            {"failure_message": "There are no existing racks within this range. "},
-            status=HTTPStatus.BAD_REQUEST,
-        )
-
-    racks_with_instances = []
-    for rack in racks:
-        rack_serializer = RackSerializer(rack)
-        instances = ITInstance.objects \
-            .filter(rack=rack.id) \
-            .order_by("elevation")
-        instances_serializer = RecursiveITInstanceSerializer(
-            instances,
-            many=True
-        )
-        rack_detail = {
-            "rack": rack_serializer.data,
-            "instances": instances_serializer.data,
-        }
-        racks_with_instances.append(rack_detail)
-
-    return JsonResponse(
-        {"racks": racks_with_instances},
-        status=HTTPStatus.OK
-    )
+    return get_rack_detailed_response(racks)
```

Comment:
refactored this into a shared util function

Reply:
b.e.a.u.t.i.f.u.l.